src/System.Collections/tests/Generic/Dictionary/DictionaryTests.cs

```diff
@@ -0,0 +1,164 @@
+﻿// Copyright (c) Justin Van Patten. All rights reserved.
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Xunit;
+
+namespace System.Collections.Tests
+{
+    public class DictionaryTests
+    {
+        [Fact]
+        public void CopyConstructorExceptions()
+        {
+            Assert.Throws<ArgumentNullException>("dictionary", () => new Dictionary<int, int>((IDictionary<int, int>)null));
+            Assert.Throws<ArgumentNullException>("dictionary", () => new Dictionary<int, int>((IDictionary<int, int>)null, null));
+            Assert.Throws<ArgumentNullException>("dictionary", () => new Dictionary<int, int>((IDictionary<int, int>)null, EqualityComparer<int>.Default));
+
+            Assert.Throws<ArgumentOutOfRangeException>("capacity", () => new Dictionary<int, int>(new NegativeCountDictionary<int, int>()));
+            Assert.Throws<ArgumentOutOfRangeException>("capacity", () => new Dictionary<int, int>(new NegativeCountDictionary<int, int>(), null));
+            Assert.Throws<ArgumentOutOfRangeException>("capacity", () => new Dictionary<int, int>(new NegativeCountDictionary<int, int>(), EqualityComparer<int>.Default));
+        }
+
+        [Theory]
+        [MemberData("CopyConstructorInt32Data")]
+        public void CopyConstructorInt32(int size, Func<int, int> keyValueSelector, Func<Dictionary<int, int>, IDictionary<int, int>> dictionarySelector, IEqualityComparer<int> comparer)
+        {
+            TestCopyConstructor(size, keyValueSelector, dictionarySelector);
```

Comment:
We're taking a comparer and using the same MemberData even though it's not being used here in TestCopyConstructor.  Doesn't that mean we're running each of these tests N times (where N is the number of comparers used in the matrix) even though the comparer isn't used?